src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -148,6 +149,11 @@ public ActiveDirectoryDomain(String name, String servers, String site, String bi
 
     @DataBoundConstructor
     public ActiveDirectoryDomain(String name, String servers, String site, String bindName, String bindPassword, TlsConfiguration tlsConfiguration) {
+        // Gives exception if an insecure certificate is used in FIPS mode.
+        if (isFipsNonCompliant(tlsConfiguration.equals(TlsConfiguration.TRUST_ALL_CERTIFICATES))) {
```

Comment:
this is the line that is causing the CI failures. If you receive a null value, then that's a NPE. The good practice is always to change the order of the condition
```suggestion
        if (isFipsNonCompliant(TlsConfiguration.TRUST_ALL_CERTIFICATES.equals(tlsConfiguration))) {
```